### DISCUSSION TITLE
How to debug Ghidra ?

### DISCUSSION THREAD

**Discussion by esaulenka** [NationalSecurityAgency/ghidra#5926] [Q&A] [answered]
I'm trying to run Ghidra in the Eclipse debugger.

What I did:
- pulled the latest sources
- built Ghidra with gradle
- added Eclipse extensions from latest official release (Honestly, will be better to add more visible remark that it don't compiled by common gradle task and one should do it separately. I spent more than hour here).
- built and imported Eclipse projects

So, I am able to open and edit some Ghidra parts. But when I am trying to run debug, Eclipse opens a startup Ghidra window (where you can select binary to analyze) without any 'tool'. Menu Tools -> Import Default Tools also empty.

Could anyone advice me?

I am linked Eclipse with compiled version of the Ghidra, not with source folder, is it right?
I am running it on Win10, if it matters.

**Comment by ryanmkurtz** [comment, marked as answer]
Did you follow [these ](https://github.com/NationalSecurityAgency/ghidra#advanced-development) instructions when setting up your Eclipse environment?

**Comment by esaulenka** [comment]
Sorry, probably I forgot to run `gradle buildNatives` (is it not the part of `buildGhdra` ?)
Now I see all tools there, thanks a lot.
